File: json_memory_tests/php/nested_100.php
<?php
$parent = ["val" => 0, "child" => null];

$t0 = microtime(true);

for ($i = 1; $i <= 100; $i++) {
    $parent = ["val" => $i, "child" => $parent];
}

$serialized = json_encode($parent);
$parsed = json_decode($serialized, true);

$explorer = $parsed;
for ($k = 1; $k <= 50; $k++) {
    $explorer = $explorer["child"];
}
$val_check = $explorer["val"];

$t1 = microtime(true);
$json_ms = ($t1 - $t0) * 1000.0;
echo "json_elapsed_ms: " . number_format($json_ms, 3, '.', '') . "\n";

File: json_memory_tests/python/nested_100.py
import json
import time

parent = {"val": 0, "child": None}
t0 = time.perf_counter()
for i in range(1, 100 + 1):
    parent = {"val": i, "child": parent}

serialized = json.dumps(parent)
parsed = json.loads(serialized)

explorer = parsed
for k in range(1, 50 + 1):
    explorer = explorer["child"]
val_check = explorer["val"]

t1 = time.perf_counter()
json_ms = (t1 - t0) * 1000.0
print(f"json_elapsed_ms: {json_ms:.3f}")
